File: src/content/3.10/code/scala/snippet09.scala
def lambda[A, B, P[_, _]](P: Profunctor[P]): P[A, A] => (A => B) => P[A, B] =
  paa => f => P.dimap(identity[A])(f)(paa)

def rho[A, B, P[_, _]](P: Profunctor[P]): P[B, B] => (A => B) => P[A, B] =
  pbb => f => P.dimap(f)(identity[B])(pbb)

File: src/content/3.10/code/ocaml/snippet09.ml
module EndsEqualizer (P : Profunctor) = struct
  let lambda (paa : ('a, 'a) P.t) (f : 'a -> 'b) : ('a, 'b) P.t =
    P.dimap Fun.id f paa

  let rho (pbb : ('b, 'b) P.t) (f : 'a -> 'b) : ('a, 'b) P.t =
    P.dimap f Fun.id pbb
end
